### PR TITLE
add model compression support for  models with exclude_types feature

### PR DESCRIPTION
**Pull request by denghuilu**
### changes

- add support for exclude_types model.

### CODE REVIEW

**Comment by denghuilu** [comment]
#658 should be fixed now.

File: deepmd/utils/tabulate.py [hunk 1]
```diff
@@ -19,7 +19,8 @@ class DeepTabulate():
     """
     def __init__(self,
                  model_file : str,
-                 type_one_side : bool = False) -> None:
+                 type_one_side : bool = False,
+                 exclude_types : List[int] = []) -> None:
```
Comment (njzjz) [inline deepmd/utils/tabulate.py:23]:
```suggestion
                 exclude_types : List[List[int]] = []) -> None:
```

File: deepmd/utils/tabulate.py [hunk 2]
```diff
@@ -29,10 +30,15 @@ def __init__(self,
                 The frozen model
         type_one_side
                 Try to build N_types tables. Otherwise, building N_types^2 tables
+        exclude_types : list[int]
```
Comment (njzjz) [inline deepmd/utils/tabulate.py:33]:
```suggestion
        exclude_types : list[list[int]]
```

**Review by njzjz — CHANGES_REQUESTED** [review]

**Review by njzjz — APPROVED** [review]

File: deepmd/utils/tabulate.py [hunk 3]
```diff
@@ -106,27 +119,27 @@ def build(self,
...
-                        tt = stride1
-                    hh = vv[jj + 1][kk] - vv[jj][kk]
-                    self.data[net][jj][kk * 6 + 0] = vv[jj][kk]
-                    self.data[net][jj][kk * 6 + 1] = dd[jj][kk]
-                    self.data[net][jj][kk * 6 + 2] = 0.5 * d2[jj][kk]
-                    self.data[net][jj][kk * 6 + 3] = (1 / (2 * tt * tt * tt)) * (20 * hh - (8 * dd[jj + 1][kk] + 12 * dd[jj][kk]) * tt - (3 * d2[jj][kk] - d2[jj + 1][kk]) * tt * tt)
-                    self.data[net][jj][kk * 6 + 4] = (1 / (2 * tt * tt * tt * tt)) * (-30 * hh + (14 * dd[jj + 1][kk] + 16 * dd[jj][kk]) * tt + (3 * d2[jj][kk] - 2 * d2[jj + 1][kk]) * tt * tt)
-                    self.data[net][jj][kk * 6 + 5] = (1 / (2 * tt * tt * tt * tt * tt)) * (12 * hh - 6 * (dd[jj + 1][kk] + dd[jj][kk]) * tt + (d2[jj + 1][kk] - d2[jj][kk]) * tt * tt)
-            self.data[net]
+            if self.type_one_side or ((int(ii / self.ntypes), int(ii % self.ntypes)) not in self.exclude_types):
```
Comment (njzjz) [inline deepmd/utils/tabulate.py:122]: In fact you can use `ii // self.ntypes` other than `int(ii / self.ntypes)`.

**Review by njzjz — APPROVED** [review]

**Pull request merged**